### ISSUE TITLE
Deprecated Rails Method passing an instance of ActiveRecord::Base to 'find'

### ISSUE THREAD

**Issue by bpmich** [Freika/dawarich#152]
**Describe the bug**
Following an import of my Google Takeout, all jobs in sidekiq are showing as failing.

**Version**
freikin/dawarich:latest [0.9.9] (https://hub.docker.com/layers/freikin/dawarich/0.9.9/images/sha256-374de3389014da0db0a80320d5fdcdf0495180e0070f5ef70fa2cbc79f89d924?context=explore)

**To Reproduce**
1. perform clean install of dawarich
2. upload Records.json from Google Takeout to dawarich at /var/app/tmp
3. import Records.json using the following command: 
`bundle exec rake import:big_file['tmp/Records.json','jvp3.14@gmail.com']`
4. wait for upload to complete and open https://dawarich.domain.tld/sidekiq
5. Observe all jobs listed as failures

**Expected behavior**
5. Observe NO jobs listed as failures

**Screenshots**
![image](https://github.com/user-attachments/assets/e02dc195-d558-4ad6-84fb-210df776e600)
![image](https://github.com/user-attachments/assets/c9d734aa-255a-4837-81ba-0c31a493232d)

**Logs**
`WARN: ArgumentError: You are passing an instance of ActiveRecord::Base to 'find'. Please pass the id of the object by calling '.id'.`

**Additional context**
Please see this Github issue for Rails I found regarding the deprecation of this method. I don't understand Rails much, but it seems relevant: https://github.com/rails/rails/issues/21945

**Comment by Freika** [comment]
Without a full stacktrace of a moment where you got this warning I won't be able to find the cause, unfortunately. Also, a warning is not an indication of a bug, and failed sidekiq jobs are related to something else, some errors were raised during the importing process. To debug that I'll need to have a log of an error raised and also an example of a file you're trying to export. The one with anonymized coordinates but preserved file structure will do.

Thanks!

**Comment by bpmich** [comment]
Please see the expanded log from sidekiq:

```
2024-08-05T01:58:27.826Z pid=91 tid=22j7 class=ImportGoogleTakeoutJob jid=4b33a5ced5cb859837d28144 INFO: start
D, [2024-08-05T01:58:27.832800 #91] DEBUG -- :   Import Load (2.3ms)  SELECT "imports".* FROM "imports" WHERE "imports"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
I, [2024-08-05T01:58:27.836752 #91]  INFO -- : Performing ImportGoogleTakeoutJob (Job ID: 6c8ea9a3-49b3-4f30-8aff-c398b244d329) from Sidekiq(imports) enqueued at 2024-08-05T01:57:04.316321127Z with arguments: #<GlobalID:0x00007f19e862a030 @uri=#<URI::GID gid://dawarich/Import/2>>, "{\"latitudeE7\":414636804,\"longitudeE7\":-871663261,\"accuracy\":30,\"activity\":[{\"activity\":[{\"type\":\"STILL\",\"confidence\":100}],\"timestamp\":\"2013-01-20T06:30:32.174Z\"}],\"source\":\"WIFI\",\"deviceTag\":899877829,\"timestamp\":\"2013-01-20T06:30:46.834Z\"}"
E, [2024-08-05T01:58:27.838284 #91] ERROR -- : Error performing ImportGoogleTakeoutJob (Job ID: 6c8ea9a3-49b3-4f30-8aff-c398b244d329) from Sidekiq(imports) in 9.49ms: ArgumentError (You are passing an instance of ActiveRecord::Base to `find`. Please pass the id of the object by calling `.id`.):
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:508:in `find_one'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:499:in `find_with_ids'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:92:in `find'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/querying.rb:23:in `find'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/core.rb:250:in `find'
/var/app/app/jobs/import_google_takeout_job.rb:8:in `perform'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:69:in `block in _perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/i18n-1.14.5/lib/i18n.rb:351:in `with_locale'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/core_ext/time/zones.rb:65:in `use_zone'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:141:in `run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:68:in `_perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:32:in `_perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:52:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:26:in `block in perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/railties/job_runtime.rb:13:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:40:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications.rb:206:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications/instrumenter.rb:58:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications.rb:206:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:39:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/railties/job_runtime.rb:11:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:26:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:18:in `block in perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:27:in `tag_logger'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:18:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:30:in `block in execute'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/railtie.rb:67:in `block (4 levels) in <class:Railtie>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:77:in `block in wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:74:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/railtie.rb:66:in `block (3 levels) in <class:Railtie>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:141:in `run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:28:in `execute'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/queue_adapters/sidekiq_adapter.rb:70:in `perform'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:210:in `execute_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:180:in `block (4 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:180:in `traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:183:in `block in traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/metrics/tracking.rb:26:in `track'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/metrics/tracking.rb:126:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:182:in `traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:173:in `invoke'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:179:in `block (3 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:140:in `block (6 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_retry.rb:113:in `local'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:139:in `block (5 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/rails.rb:16:in `block in call'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:77:in `block in wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:74:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/rails.rb:15:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:135:in `block (4 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:271:in `stats'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:130:in `block (3 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_logger.rb:13:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:129:in `block (2 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_retry.rb:80:in `global'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:128:in `block in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_logger.rb:39:in `prepare'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:127:in `dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:178:in `block (2 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:177:in `handle_interrupt'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:177:in `block in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:176:in `handle_interrupt'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:176:in `process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:82:in `process_one'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:72:in `run'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/component.rb:10:in `watchdog'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/component.rb:19:in `block in safe_thread'
2024-08-05T01:58:27.839Z pid=91 tid=21x7 class=ImportGoogleTakeoutJob jid=426598e4989fedf701e249b5 elapsed=0.807 INFO: fail
2024-08-05T01:58:27.839Z pid=91 tid=21x7 WARN: {"context":"Job raised exception","job":{"retry":false,"queue":"imports","wrapped":"ImportGoogleTakeoutJob","args":[{"job_class":"ImportGoogleTakeoutJob","job_id":"6c8ea9a3-49b3-4f30-8aff-c398b244d329","provider_job_id":null,"queue_name":"imports","priority":null,"arguments":[{"_aj_globalid":"gid://dawarich/Import/2"},"{\"latitudeE7\":414636804,\"longitudeE7\":-871663261,\"accuracy\":30,\"activity\":[{\"activity\":[{\"type\":\"STILL\",\"confidence\":100}],\"timestamp\":\"2013-01-20T06:30:32.174Z\"}],\"source\":\"WIFI\",\"deviceTag\":899877829,\"timestamp\":\"2013-01-20T06:30:46.834Z\"}"],"executions":0,"exception_executions":{},"locale":"en","timezone":"America/New_York","enqueued_at":"2024-08-05T01:57:04.316321127Z","scheduled_at":null}],"class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","jid":"426598e4989fedf701e249b5","created_at":1722823024.316383,"enqueued_at":1722823024.3164263}}
2024-08-05T01:58:27.841Z pid=91 tid=21x7 WARN: ArgumentError: You are passing an instance of ActiveRecord::Base to `find`. Please pass the id of the object by calling `.id`.
2024-08-05T01:58:27.842Z pid=91 tid=21x7 WARN: app/jobs/import_google_takeout_job.rb:8:in `perform'
2024-08-05T01:58:27.848Z pid=91 tid=22jv class=ImportGoogleTakeoutJob jid=087716aadcf5e226c91cbae0 INFO: start
D, [2024-08-05T01:58:27.924255 #91] DEBUG -- :   Import Load (3.1ms)  SELECT "imports".* FROM "imports" WHERE "imports"."id" = $1 LIMIT $2  [["id", 2], ["LIMIT", 1]]
I, [2024-08-05T01:58:27.930496 #91]  INFO -- : Performing ImportGoogleTakeoutJob (Job ID: c953daea-9c65-4829-9902-b489637bb4a3) from Sidekiq(imports) enqueued at 2024-08-05T01:57:04.317809163Z with arguments: #<GlobalID:0x00007f19e5668c70 @uri=#<URI::GID gid://dawarich/Import/2>>, "{\"latitudeE7\":414636714,\"longitudeE7\":-871662860,\"accuracy\":30,\"activity\":[{\"activity\":[{\"type\":\"STILL\",\"confidence\":100}],\"timestamp\":\"2013-01-20T06:33:32.672Z\"}],\"source\":\"WIFI\",\"deviceTag\":899877829,\"timestamp\":\"2013-01-20T06:34:46.888Z\"}"
E, [2024-08-05T01:58:27.932873 #91] ERROR -- : Error performing ImportGoogleTakeoutJob (Job ID: c953daea-9c65-4829-9902-b489637bb4a3) from Sidekiq(imports) in 16.91ms: ArgumentError (You are passing an instance of ActiveRecord::Base to `find`. Please pass the id of the object by calling `.id`.):
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:508:in `find_one'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:499:in `find_with_ids'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/relation/finder_methods.rb:92:in `find'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/querying.rb:23:in `find'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/core.rb:250:in `find'
/var/app/app/jobs/import_google_takeout_job.rb:8:in `perform'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:69:in `block in _perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/i18n-1.14.5/lib/i18n.rb:351:in `with_locale'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/core_ext/time/zones.rb:65:in `use_zone'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:141:in `run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:68:in `_perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:32:in `_perform_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:52:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:26:in `block in perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/railties/job_runtime.rb:13:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:40:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications.rb:206:in `block in instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications/instrumenter.rb:58:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/notifications.rb:206:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:39:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.4/lib/active_record/railties/job_runtime.rb:11:in `instrument'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/instrumentation.rb:26:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:18:in `block in perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:27:in `tag_logger'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/logging.rb:18:in `perform_now'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:30:in `block in execute'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:121:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/railtie.rb:67:in `block (4 levels) in <class:Railtie>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:77:in `block in wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/execution_wrapper.rb:88:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:74:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/railtie.rb:66:in `block (3 levels) in <class:Railtie>'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `instance_exec'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:130:in `block in run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/callbacks.rb:141:in `run_callbacks'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/execution.rb:28:in `execute'
/var/app/vendor/bundle/ruby/3.2.0/gems/activejob-7.1.3.4/lib/active_job/queue_adapters/sidekiq_adapter.rb:70:in `perform'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:210:in `execute_job'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:180:in `block (4 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:180:in `traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:183:in `block in traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/metrics/tracking.rb:26:in `track'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/metrics/tracking.rb:126:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:182:in `traverse'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/middleware/chain.rb:173:in `invoke'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:179:in `block (3 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:140:in `block (6 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_retry.rb:113:in `local'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:139:in `block (5 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/rails.rb:16:in `block in call'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:77:in `block in wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/execution_wrapper.rb:92:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.4/lib/active_support/reloader.rb:74:in `wrap'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/rails.rb:15:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:135:in `block (4 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:271:in `stats'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:130:in `block (3 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_logger.rb:13:in `call'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:129:in `block (2 levels) in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_retry.rb:80:in `global'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:128:in `block in dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/job_logger.rb:39:in `prepare'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:127:in `dispatch'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:178:in `block (2 levels) in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:177:in `handle_interrupt'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:177:in `block in process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:176:in `handle_interrupt'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:176:in `process'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:82:in `process_one'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/processor.rb:72:in `run'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/component.rb:10:in `watchdog'
/var/app/vendor/bundle/ruby/3.2.0/gems/sidekiq-7.2.4/lib/sidekiq/component.rb:19:in `block in safe_thread'
2024-08-05T01:58:27.937Z pid=91 tid=221f class=ImportGoogleTakeoutJob jid=fc732765cc1491843217c50e elapsed=0.83 INFO: fail
2024-08-05T01:58:27.937Z pid=91 tid=221f WARN: {"context":"Job raised exception","job":{"retry":false,"queue":"imports","wrapped":"ImportGoogleTakeoutJob","args":[{"job_class":"ImportGoogleTakeoutJob","job_id":"c953daea-9c65-4829-9902-b489637bb4a3","provider_job_id":null,"queue_name":"imports","priority":null,"arguments":[{"_aj_globalid":"gid://dawarich/Import/2"},"{\"latitudeE7\":414636714,\"longitudeE7\":-871662860,\"accuracy\":30,\"activity\":[{\"activity\":[{\"type\":\"STILL\",\"confidence\":100}],\"timestamp\":\"2013-01-20T06:33:32.672Z\"}],\"source\":\"WIFI\",\"deviceTag\":899877829,\"timestamp\":\"2013-01-20T06:34:46.888Z\"}"],"executions":0,"exception_executions":{},"locale":"en","timezone":"America/New_York","enqueued_at":"2024-08-05T01:57:04.317809163Z","scheduled_at":null}],"class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","jid":"fc732765cc1491843217c50e","created_at":1722823024.3178713,"enqueued_at":1722823024.3179169}}
2024-08-05T01:58:27.939Z pid=91 tid=221f WARN: ArgumentError: You are passing an instance of ActiveRecord::Base to `find`. Please pass the id of the object by calling `.id`.
2024-08-05T01:58:27.941Z pid=91 tid=221f WARN: app/jobs/import_google_takeout_job.rb:8:in `perform'
```

**Comment by Freika** [comment]
Hi @bpmich! I tested your snippet and was able to reproduce the bug, it was fixed in [0.9.11](https://github.com/Freika/dawarich/releases/tag/0.9.11)

**Comment by bpmich** [comment]
Hi @Freika, thank you for your expeditious response! I have tested 0.9.11 and 0.9.12, the issue is resolved on my end as well
